src/NJsonSchema.CodeGeneration.Tests/InheritanceSerializationTests.cs

```diff
@@ -259,5 +263,76 @@ public void JsonInheritanceConverter_is_thread_safe()
+
+            var assembly = Compile(code);
+            var type = assembly.GetType("foo.Foo");
+            if (type == null)
+            {
+                throw new Exception("Foo not found in " + String.Join(", ", assembly.GetTypes().Select(t => t.Name)));
+            }
+
+            var bar = JsonConvert.DeserializeObject(@"{""discriminator"":""bar""}", type);
+
+            //// Assert
+            Assert.Contains(@"""bar""", JsonConvert.SerializeObject(bar));
+        }
+
+        private Assembly Compile(string code)
+        {
+#if NETCOREAPP2_0
+            var coreDir = Directory.GetParent(typeof(Enumerable).GetTypeInfo().Assembly.Location);
+
+            Microsoft.CodeAnalysis.CSharp.CSharpCompilation compilation = Microsoft.CodeAnalysis.CSharp.CSharpCompilation.Create("assemblyName")
+             .WithOptions(new Microsoft.CodeAnalysis.CSharp.CSharpCompilationOptions(Microsoft.CodeAnalysis.OutputKind.DynamicallyLinkedLibrary))
+             .AddReferences(Microsoft.CodeAnalysis.MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
+                            Microsoft.CodeAnalysis.MetadataReference.CreateFromFile(typeof(JsonConvert).Assembly.Location),
+                            Microsoft.CodeAnalysis.MetadataReference.CreateFromFile(typeof(GeneratedCodeAttribute).Assembly.Location),
+                            Microsoft.CodeAnalysis.MetadataReference.CreateFromFile(coreDir.FullName + Path.DirectorySeparatorChar + "System.Runtime.dll"),
+                            Microsoft.CodeAnalysis.MetadataReference.CreateFromFile(coreDir.FullName + Path.DirectorySeparatorChar + "System.Dynamic.Runtime.dll"),
+                            Microsoft.CodeAnalysis.MetadataReference.CreateFromFile(coreDir.FullName + Path.DirectorySeparatorChar + "System.IO.dll"),
+                            Microsoft.CodeAnalysis.MetadataReference.CreateFromFile(coreDir.FullName + Path.DirectorySeparatorChar + "System.Linq.Expressions.dll"),
+                            Microsoft.CodeAnalysis.MetadataReference.CreateFromFile(coreDir.FullName + Path.DirectorySeparatorChar + "System.Runtime.Extensions.dll"))
+             .AddSyntaxTrees(Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTree.ParseText(code));
+
+            using (var stream = new MemoryStream())
+            {
+                var result = compilation.Emit(stream);
+
+                if (!result.Success)
+                {
+                    throw new Exception(String.Join(", ", result.Diagnostics
+                        .Where(diagnostic => diagnostic.IsWarningAsError || diagnostic.Severity == Microsoft.CodeAnalysis.DiagnosticSeverity.Error)
+                        .Select(d => d.Location.GetLineSpan().StartLinePosition + " - " + d.GetMessage())) + "\n" + code);
+                }
+
+                return Assembly.Load(stream.GetBuffer());
+            }
+#endif
+
+#if NET451
+            var provider = new CSharpCodeProvider();

```

Comment:
This needs to be fully qualified as `Microsoft.CSharp.CSharpCodeProvider`, or a `using` needs to be added. The code doesn't build as-is.

Reply:
Thanks for finding this problem

Reply:
I'm sorry I missed that.